### PR TITLE
debug enhancements in subadminmiddleware.php

### PR DESCRIPTION
**Pull request by chludwig**
Displays the exception message on the 403 error page. Very usefull (not only for newbies) to debug, becaus a 403 page without any message or log file entry is very hard to interprete. In addition a logfile entry would be a nice option.

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
@chludwig Thanks a lot for your contribution!
Contributions to the core repo require a signed contributors agreement http://owncloud.org/about/contributor-agreement/

Alternatively you can add a comment here where you state that this contribution is MIT licensed.

Some more details about out pull request workflow can be found here: http://owncloud.org/code-reviews-on-github/

File: settings/middleware/subadminmiddleware.php [hunk 1]
```diff
@@ -71,7 +71,11 @@ public function beforeController($controller, $methodName) {
 	 * @return TemplateResponse
 	 */
 	public function afterException($controller, $methodName, \Exception $exception) {
-		$response = new TemplateResponse('core', '403', array(), 'guest');
+		$message = '';
+		if (\OC::$server->getConfig()->getSystemValue('debug', false)) {
```
Comment (DeepDiver1975) [inline settings/middleware/subadminmiddleware.php]: the config object needs to be injected via the ctor

File: settings/middleware/subadminmiddleware.php [hunk 2]
```diff
@@ -71,7 +71,11 @@ public function beforeController($controller, $methodName) {
...
+			$message = $exception->getMessage();
+		}
+		$response = new TemplateResponse('core', '403', array('file' => $message), 'guest');
```
Comment (DeepDiver1975) [inline settings/middleware/subadminmiddleware.php]: I'd prefer to write the exception to the log instead of throwing in to the user

**Comment by LukasReschke** [comment]
> Displays the exception message on the 403 error page. Very usefull (not only for newbies) to debug, becaus a 403 page without any message or log file entry is very hard to interprete. In addition a logfile entry would be a nice option.

I'd vote for something like https://github.com/master/core/commit/22e724e8292315f600b57a30716e26c1f1c04f66 instead. 

@chludwig Mind adjusting your PR?

**Comment by chludwig** [comment]
My contribution is MIT licensed.

@LukasReschke @DeepDiver1975 For me it is hard to make your proposed changes, maybe someone else can help here. Thank you.

**Comment by MorrisJobke** [comment]
> I'd vote for something like master/core@22e724e instead.

@LukasReschke This is a dead link :see_no_evil:

**Comment by PVince81** [comment]
@LukasReschke can you clarify ?

**Comment by LukasReschke** [comment]
Replaced by https://github.com/owncloud/core/pull/22551

**Pull request closed without merging**